flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/mailbox/MailboxProcessor.java

```diff
@@ -152,7 +160,7 @@ public void runMailboxLoop() throws Exception {
 		final MailboxController defaultActionContext = new MailboxController(this);
 
 		while (processMail(localMailbox)) {
-			mailboxDefaultAction.runDefaultAction(defaultActionContext);
+			mailboxDefaultAction.runDefaultAction(defaultActionContext); // currently, lock is acquired inside each action
```

Comment:
Do you plan to change that? Is that even possible? Or is it the idea to only acquire it for legacy source tasks and we just accept the performance implications?

Reply:
Probably, I misworded the comment (updated).
The lock is acquired inside `default action` as needed, not in a uniform fashion.

Reply:
Probably, I misworded the comment (updated).
The lock is acquired inside `default action` as needed, not in a uniform fashion.

Reply:
@AHeise because default actions are only used in network (non source) tasks, once we drop the `StreamTask#getCheckpointLock` method and `checkpointLock` will be used only in the source tasks, there will be no need for any `checkpointLock` usage inside the default actions ever - the current usage will be just dropped, without using any `StreamTaskActionExecutor`.